test/unittests/test_dd_synthesis_dc.cpp

```diff
@@ -76,20 +62,49 @@ TEST_P(TestDDSynthDc, GenericDDSynthesisDcTest) {
     EXPECT_TRUE(readPla(tt, fileName));
     extend(tt);
 
-    encodeHuffman(tt);
+    TruthTable const ttOri(tt); //unchanged Truth Table.
 
-    augmentWithConstants(tt);
+    //n -> No. of primary inputs.
+    //m -> No. of primary outputs.
+    //k1 -> Minimum no. of additional lines required.
+    //codewords -> Output patterns with the respective codewords.
+    //totalNoBits -> Total no. of bits required to create the circuit
+    // r -> additional variables/bits required to decode r MSB bits of the circuit.
+
+    const auto n = tt.nInputs();
+    const auto m = tt.nOutputs();
+
+    const auto [codewords, k1] = encodeHuffman(tt);
+
+    const auto totalNoBits = std::max(n, m + k1);
+    const auto r           = totalNoBits - tt.nOutputs();
+
+    augmentWithConstants(tt, totalNoBits);
 
     const auto ttDD = buildDD(tt, dd);
     EXPECT_TRUE(ttDD.p != nullptr);
 
-    DDSynthesizer synthesizer(tt.nInputs());
-    const auto&   qc = synthesizer.synthesize(ttDD, dd);
+    DDSynthesizer synthesizer(totalNoBits);
+
+    // codewords, r, m required for decoding purposes.
+    const auto& qc = synthesizer.synthesize(ttDD, codewords, r, m, dd);
+
+    //Expected Truth table
+    for (const auto& [input, output]: ttOri) {
+        auto inCube(input);
+        for (auto i = 0U; i < (totalNoBits - input.size()); i++) {
+            inCube.insertZero();
+        }
+        TruthTable::Cube outCube(output);
+        outCube.resize(totalNoBits);
+        ttExpected.try_emplace(inCube, outCube);
+    }
```

Comment:
Isn't this somewhat close to or even the same as the `augmentWithConstants` function does?
In any case, this should not be a function in the tests but a method of the truth table class.
Furthermore, it should augment an existing TT and not create yet another copy.